samples/monster_generated.rs

```diff
@@ -1,148 +1,163 @@
 // automatically generated by the FlatBuffers compiler, do not modify
 
 
+#![allow(unused_imports, dead_code)]
+
+use std::mem;
+use std::cmp::Ordering;
+
+extern crate flatbuffers;
+use self::flatbuffers::EndianScalar;
+
+#[allow(unused_imports, dead_code)]
 pub mod my_game {
-  #![allow(dead_code)]
-  #![allow(unused_imports)]
 
   use std::mem;
-  use std::marker::PhantomData;
   use std::cmp::Ordering;
 
   extern crate flatbuffers;
   use self::flatbuffers::EndianScalar;
+#[allow(unused_imports, dead_code)]
 pub mod sample {
-  #![allow(dead_code)]
-  #![allow(unused_imports)]
 
   use std::mem;
-  use std::marker::PhantomData;
   use std::cmp::Ordering;
 
   extern crate flatbuffers;
   use self::flatbuffers::EndianScalar;
 
-#[allow(non_camel_case_types)]
-#[repr(i8)]
-#[derive(Clone, Copy, PartialEq, Debug)]
-pub enum Color {
-  Red = 0,
-  Green = 1,
-  Blue = 2
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Color(pub i8);
+#[allow(non_upper_case_globals)]
+impl Color {
+  pub const ENUM_MIN: i8 = 0;
+  pub const ENUM_MAX: i8 = 2;
+  pub const Red: Self = Self(0);
+  pub const Green: Self = Self(1);
+  pub const Blue: Self = Self(2);
+  pub const ENUM_VALUES: &'static [Self] = &[
+    Self::Red,
+    Self::Green,
+    Self::Blue,
+  ];
+  /// Returns the variant's name or "" if unknown.
+  pub fn variant_name(self) -> &'static str {
+    match self {
+      Self::Red => "Red",
+      Self::Green => "Green",
+      Self::Blue => "Blue",
+      _ => "",
+    }
```

Comment:
References are null-optimized; please, use `Option<&'static str>`, and return `None` for invalid values.

Reply:
Forcing every caller to match or unwrap? That seems pretty unergonomic to me for a case that should never happen in practice

Reply:
Well, this is the idiomatic way of Rust. I may suggest adding a variant which panics, for those not wanting to deal with Option.

Reply:
Welp, you can make this function unsafe and use `_ => unsafe { std::hint::unreachable_unchecked() }`. No unwraps, only `unsafe` crawling everywhere and possibility of UB. Or you can keep this function safe, but use `_ => unreachable!()`. The code size should stay the same here, panics are `#[inline(never)]` functions and one usize wide, IIRC. One way or another, there is that branch, and nothing except `unreachable_unchecked` would delete this branch.

Reply:
Done, we're using Option.